packages/medusa/src/utils/validator.ts

```diff
@@ -1,6 +1,34 @@
 import { ClassConstructor, plainToInstance } from "class-transformer"
 import { validate, ValidationError, ValidatorOptions } from "class-validator"
 import { MedusaError } from "medusa-core-utils"
+import { Constructor } from "@medusajs/types"
+
+const extendedValidators: Map<string, Constructor<any>> = new Map()
+
+/**
+ * When overriding a validator, you can register it to be used instead of the original one.
+ * For example, the place where you are overriding the core validator, you can call this function
+ * @example
+ * ```ts
+ * // /src/api/routes/admin/products/create-product.ts
+ * import { registerOverriddenValidators } from "@medusajs/medusa"
+ * import { AdminPostProductsReq as MedusaAdminPostProductsReq } from "@medusajs/medusa/dist/api/routes/admin/products/create-product"
+ * import { IsString } from "class-validator"
+ *
+ * class AdminPostProductsReq extends MedusaAdminPostProductsReq {
+ *    @IsString()
+ *    test: string
+ * }
+ *
+ * registerOverriddenValidators(AdminPostProductsReq)
+ * ```
+ * @param extendedValidator
+ */
+export function registerOverriddenValidators(
```

Comment:
Looking forward to the merge, then we can finally remove the loader for patching `@medusajs/medusa/dist/utils/validator`